etcd/etcd.go

```diff
@@ -127,3 +131,21 @@ func (r *EtcdAdapter) Refresh(service *bridge.Service) error {
 func (r *EtcdAdapter) Services() ([]*bridge.Service, error) {
 	return []*bridge.Service{}, nil
 }
+
+func (r *EtcdAdapter) garbageCollection(rootDir string, dirPath string) {
+	for rootDir != dirPath {
+		var err error
+		if r.client != nil {
+			_, err = r.client.DeleteDir(dirPath)
+		} else {
+			_, err = r.client2.DeleteDir(dirPath)
+		}
+
+		if err != nil {
+			return
+		}
+
+		dirPath, _ := path.Split(dirPath)
```

Comment:
Correction:

    -dirPath, _ := path.Split(dirPath)
    +dirPath, _ = path.Split(dirPath)

The `:=` will case you to create a new variable dirPath inside the loop. As a consequence, dirPath will be reset to it's initial value every time you start a new loop